astlib/migrate_501_500.ml

```diff
@@ -262,6 +318,44 @@ and copy_value_binding :
             (List.rev locally_abstract_univars)
         in
         (pvb_pat, pvb_expr)
+    | Some (Pvc_constraint { locally_abstract_univars = []; typ }), _ ->
+        let typ = copy_core_type typ in
+        let pvb_pat =
+          {
+            pvb_pat with
+            ppat_attributes = [];
+            ppat_desc = Ast_500.Parsetree.Ppat_constraint (pvb_pat, typ);
+          }
+        in
+        (pvb_pat, pvb_expr)
```

Comment:
To be completely faithful at the parsetree level, the expression part should have a constraint added to the `pvb_expr` part when `typ` is not a `Ptyp_poly` .

However, `let (x:typ) = y` and `let (x:typ) = (y:typ)` are equivalent after typechecking, so this this is only visible at the parsetree level.